templates/common/infra/terraform/core/database/postgresql/postgresql.tf

```diff
@@ -0,0 +1,78 @@
+terraform {
+  required_providers {
+    azurerm = {
+      version = "~>3.47.0"
+      source  = "hashicorp/azurerm"
+    }
+    azurecaf = {
+      source  = "aztfmod/azurecaf"
+      version = "~>1.2.24"
+    }
+  }
+}
+# ------------------------------------------------------------------------------------------------------
+# Deploy PostgreSQL Server
+# ------------------------------------------------------------------------------------------------------
+resource "azurecaf_name" "psql" {
+  name          = var.resource_token
+  resource_type = "azurerm_postgresql_flexible_server"
+  random_length = 0
+  clean_input   = true
+}
+
+data "azurerm_client_config" "current" {}
+
+locals {
+  principal_type = "${var.client_id == "" ? "User" : "ServicePrincipal"}"
```

Comment:
This isn't either/or.  If client_id is provided use it and assign roles.  Also, assign SP roles for the service.

Reply:
This is not for role assignment scenario. It's used for [PostgreSQL AAD authentication setting](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/postgresql_flexible_server_active_directory_administrator#principal_type), choose User or ServicePrincipal as a principal type.